File: solutions/Virtual-Assistant/src/csharp/tests/Microsoft.Bot.Solutions.Tests/Skills/Fakes/FakeSkill/Dialogs/Shared/Resources/SharedResponses.cs
// https://docs.microsoft.com/en-us/visualstudio/modeling/t4-include-directive?view=vs-2017
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
using System.IO;
using System.Runtime.CompilerServices;
using Microsoft.Bot.Solutions.Dialogs;

namespace FakeSkill.Dialogs.Shared.Resources
{
    /// <summary>
    /// Contains bot responses.
    /// </summary>
    public static class SharedResponses
    {
        private static readonly ResponseManager _responseManager;

        static SharedResponses()
        {
            var dir = Path.GetDirectoryName(typeof(SharedResponses).Assembly.Location);
            var resDir = Path.Combine(dir, @"Skills\Fakes\FakeSkill\Dialogs\Shared\Resources");
            _responseManager = new ResponseManager(resDir, "SharedResponses");
        }

        // Generated accessors
        public static BotResponse DidntUnderstandMessage => GetBotResponse();

        public static BotResponse CancellingMessage => GetBotResponse();

        public static BotResponse NoAuth => GetBotResponse();

        public static BotResponse AuthFailed => GetBotResponse();

        public static BotResponse ActionEnded => GetBotResponse();

        public static BotResponse ErrorMessage => GetBotResponse();

        private static BotResponse GetBotResponse([CallerMemberName] string propertyName = null)
        {
            return _responseManager.GetBotResponse(propertyName);
        }
    }
}

File: templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/skill/templates/sample-skill/src/responses/shared/sharedResponses.ts
/**
 * Copyright(c) Microsoft Corporation.All rights reserved.
 * Licensed under the MIT License.
 */

import { IResponseIdCollection } from 'botbuilder-solutions';
import { join } from 'path';

/**
 * Contains bot responses.
 */
export class SharedResponses implements IResponseIdCollection {
    // Generated accessors
    public name: string = SharedResponses.name;
    public static pathToResource?: string = join(__dirname, 'resources');
    public static readonly didntUnderstandMessage   : string = 'DidntUnderstandMessage';
    public static readonly cancellingMessage : string = 'CancellingMessage';
    public static readonly noAuth : string = 'NoAuth';
    public static readonly authFailed : string = 'AuthFailed';
    public static readonly actionEnded : string = 'ActionEnded';
    public static readonly errorMessage : string = 'ErrorMessage';
}
